servicetalk-opentelemetry-http/docs/modules/ROOT/pages/index.adoc

```diff
@@ -0,0 +1,220 @@
+----
+HttpServerBuilder.forAddress(localAddress(0))
+    // IMPORTANT: Use non-offloading filter for proper context propagation
+    .appendNonOffloadingServiceFilter(new OpenTelemetryHttpServiceFilter.Builder()
+        .capturedRequestHeaders(Arrays.asList("user-agent", "x-request-id"))
+        .capturedResponseHeaders(Arrays.asList("content-type"))
+        .build())
+
+    // Add request draining AFTER OpenTelemetry filter
+    .appendNonOffloadingServiceFilter(HttpRequestAutoDrainingServiceFilter.INSTANCE)
+
+    .listen(service);
+----
+
+== Advanced Configuration
+
+=== Capturing Custom Headers
+
+[source,java]
+----
+// Capture specific request and response headers as span attributes
+OpenTelemetryHttpRequesterFilter filter = new OpenTelemetryHttpRequesterFilter.Builder()
+    .componentName("my-client")
+    .capturedRequestHeaders(Arrays.asList("x-custom-header", "authorization"))
+    .capturedResponseHeaders(Arrays.asList("x-response-id", "content-type"))
+    .build();
+----
+
+== Filter Ordering Guidelines
+
+Proper filter ordering is crucial for OpenTelemetry tracing to work correctly with other ServiceTalk features.
+
+=== Server Filters
+
+For servers, use non-offloading filters for proper context propagation and pay special attention to filter ordering:
+
+[source,java]
+----
+HttpServerBuilder.forAddress(localAddress(0))
+    // OpenTelemetry filter MUST be first for proper context propagation
+    .appendNonOffloadingServiceFilter(new OpenTelemetryHttpServiceFilter.Builder()
+        .build())
+
+    // Request draining filter MUST come after OpenTelemetry filter
+    // This ensures tracing information is captured for auto-drained requests
+    .appendNonOffloadingServiceFilter(HttpRequestAutoDrainingServiceFilter.INSTANCE)
+
+    // Exception mapping should come after tracing to ensure correct status codes
+    .appendServiceFilter(HttpExceptionMapperServiceFilter.create())
+
+    // Other filters can follow
+    .listen(service);
+----
+
+**Important:** The `HttpRequestAutoDrainingServiceFilter` must be placed *after* the OpenTelemetry filter. This is critical for server-side tracing accuracy, particularly for requests like GET where the request body is typically ignored. If auto-draining occurs before the OpenTelemetry filter processes the request, tracing information may be incomplete or incorrect.
+
+=== Client Filters
+
+When using multiple filters on clients, place the OpenTelemetry filter appropriately in the chain:
+
+[source,java]
+----
+HttpClient client = HttpClients.forSingleAddress("example.com", 80)
+    // Logging or other observability filters before tracing
+    .appendClientFilter(loggingFilter)
+
+    // OpenTelemetry filter for span creation and context propagation
+    .appendClientFilter(new OpenTelemetryHttpRequesterFilter.Builder()
+        .componentName("my-client")
+        .build())
+
+    // Retry and other resilience filters after tracing
+    .appendClientFilter(retryFilter)
+    .build();
```

Comment:
Consider demonstrating that users can add another filter at the connection level and describe what it will give them